### PR TITLE
3561 refactor api v1 synchronizations

### PR DESCRIPTION
**Pull request by juanignaciosl**
@Kartones @rafatower CR this closing #3561, please.
@javisantana take a look at the JS at least, please.

I've only seen usage of `api/v1_1/synchronizations/:id` endpoint. I've not been able to trigger neither `api/v1_1/synchronizations` nor `api/v1_1/synchronizations/:id/sync_now `. I've created a new sync table, opened the dashboard, opened an sync table, forced sync and the only GET to sync endpoints is the :id one. This is consistent with [this Kibana search](http://ela01.useb.cartodb.net/#/dashboard/elasticsearch/sync%20requests). Could you @zenitraM please confirm that's the only endpoint used? In that case, do you all think I should remove the other endpoints? (see `routes.rb` changes).

### CODE REVIEW

File: spec/requests/api/json/synchronizations_controller_shared_examples.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      }
+
+      post "/api/v1/synchronizations?api_key=#{@api_key}", payload.to_json, @headers
+      id = JSON.parse(last_response.body).fetch('id')
+
+      get "/api/v1/synchronizations/#{id}/sync_now?api_key=#{@api_key}", nil, @headers
+      last_response.status.should == 200
+
+      response = JSON.parse(last_response.body)
+      response.fetch('state').should == 'created'
```
Comment (Kartones) [inline spec/requests/api/json/synchronizations_controller_shared_examples.rb]: Maybe would be nice to improve this test by calling the logic that performs the sync and check it goes to state success or fail, but maybe is out of the scope as test is not trivial and will require stubbing things and faking URL requests...

**Comment by Kartones** [comment]
uhmm, @xavijam isn't `api/v1_1/synchronizations/:id/sync_now` the endpoint for doing a "Sync now" from the ui?

If so, should be triggered if the user has the flag. Else might be just used from the rake that forces sync_now calls, I don't remember right now without checking.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3578/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3577/))

**Comment by javisantana** [comment]
js => :es:

**Comment by xavijam** [comment]
So, sync_now:

https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/models/synchronization.js#L60

**Comment by juanignaciosl** [comment]
@xavijam yep, but that's a PUT, only GETs are being moved.

**Comment by xavijam** [comment]
So, what is your proposal? or what should we do it (in frontend)?

**Comment by juanignaciosl** [comment]
No, I think you don't have to do anything, @xavijam .

**Comment by Kartones** [comment]
If not used... then fine by me removing the GET version.

Apart from that , :+1:

**Comment by xavijam** [comment]
:+1:

**Comment by juanignaciosl** [comment]
I guess I got a :+1: from @Kartones , still waiting for @rafatower

**Comment by rafatower** [comment]
:+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3642/))

**Comment by Kartones** [comment]
:+1:  that SSL forcing

**Pull request merged**